lesson/outline/outline-template.md

```diff
@@ -1,104 +1,323 @@
+3. Demonstrate the different outcomes of the five selected matchers individually so, while encouraging the learner to visit documentation for the full list. 
 
 
 
 ### Checkpoints Summary
 
-_Check out the [content standards](http://curriculum-documentation.codecademy.com/Content-Standards/checkpoint/) for guidance on writing narratives for exercises._
 
-1. Checkpoint one
-2. Checkpoint two
+Using the previously created empty test function in the myproject.test.js file. 
+1. Write simple unit tests with .toBe
+2. Run test to verify
+3. Write simple unit tests with .not
+4. Run test to verify
+5. Write simple unit tests with .toBeTruthy
+6. Run test to verify
+7. Write simple unit tests with .toBeDefined
+8. Run test to verify
+7. Write simple unit tests with .toContain
+8. Run test to verify
+ 
+
 
 #### What is the purpose of these checkpoints?
 
+- The learner will write their first unit tests 
 
 
 ### What would you like to have in the workspace for this exercise? Share your plan below.
 
-_Check out the [content standards](http://curriculum-documentation.codecademy.com/Content-Standards/workspaces/) for guidance on writing narratives for exercises._
+- I.E. text editor and output terminal
 
-<hr>
 
-## Exercise # 3: _Insert exercise title here_
+## Exercise # 5: Testing Async Code in jest: Part 1
+
+### Which course outcomes will be covered by this exercise?
+
+1. Learners will be able to understand and solve false positives in jest using `done()`
+
+
+### Narrative Summary
+
+1. Begin by analyzing an api call that because of poor jest syntax, leads to a false positive, prematurely returning a test result.
+2. Explain why the test result returned as it did and how jest works with async code
+3. Re-write code to function properly
+Use done()
+
+
+### Checkpoints Summary
+
+Using a rest api call. 
+1. Rewrite test to use done(), using above example as reference
+
+ 
+#### What is the purpose of these checkpoints?
+
+- The learner will be able to write unit tests to analyze async code using done() 
+
+
+### What would you like to have in the workspace for this exercise? Share your plan below.
+
+- I.E. text editor and output terminal
+
+
+## Exercise # 6: Testing Async Code in jest: Part 2
 
 ### Which course outcomes will be covered by this exercise?
 
-1. Learners will be able to...
+1. Learners will be able to understand and solve false positives in jest using `.resolves`, `.rejects` and async/await. 
+
 
 ### Narrative Summary
 
-_Check out the [content standards](http://curriculum-documentation.codecademy.com/Content-Standards/narrative/) for guidance on writing narratives for exercises._
+1. Explain that there are two other ways to solve the false positive issue by introducing and explaining `.resolves`, `.rejects` and async/await
+2. Re-write code to function properly using `.resolves` and `.rejects`, explaining the structure and syntax
+3.  Re-write code to function properly using async/await, explaining the structure and syntax
```

Comment:
Make it explicit that this approach is used to test functions that return Promises. 

From my understanding you need to use `async`/`await` in order to use `.resolves` and `.rejects` so they aren't actually two _different_ ways but part of the same way to deal with Promises.

Reply:
In the [docs](https://jestjs.io/docs/asynchronous) it talks about using `.resolves` and `.rejects` matchers and you can either:
1. return the expect statement:

```js
test('the data is peanut butter', () => {
  return expect(fetchData()).resolves.toBe('peanut butter');
});
```

or 2. use `async` and `await`

```js
test('the data is peanut butter', async () => {
  await expect(fetchData()).resolves.toBe('peanut butter');
});
```

I think you can cover either option but originally I had thought that using `async`/`await` is clearer